src/main/kotlin/org/rust/ide/refactoring/extractFunction/RsExtractFunctionHandler.kt

```diff
@@ -75,6 +71,27 @@ class RsExtractFunctionHandler : RefactoringActionHandler {
         }
     }
 
+    private fun findExistingInherentImpl(traitImpl: RsImplItem): RsImplItem? {
+        check(traitImpl.traitRef != null)
+        val traitCachedImpl = RsCachedImplItem.forImpl(traitImpl)
+        return (traitImpl.parent as? RsItemsOwner)
+            ?.childrenOfType<RsImplItem>()
+            ?.firstOrNull {
+                it.traitRef == null
+                    && RsCachedImplItem.forImpl(it).typeAndGenerics == traitCachedImpl.typeAndGenerics
```

Comment:
I'm not sure but maybe `org.rust.lang.core.resolve.RsCachedImplItem#isValid` needs to be checked.
For example, consider the situation when there are several `impl` blocks with `cfg` attributes:
```Rust
impl Bar for S {
    fn foo(&self) {
        // I'd expect the code to be extracted into cfg-enabled impl for `unix`
    }
}

#[cfg(windows)] // disabled
impl S {}

#[cfg(unix)] // enabled
impl S {}
```